internal/apps/dop/providers/issue/dao/issue_property.go

```diff
@@ -62,6 +65,100 @@ func (client *DBClient) UpdateIssueProperty(property *IssueProperty) error {
 	return client.Save(property).Error
 }
 
+func (client *DBClient) CleanOrgCustomFileds(orgID int64, propertyNames []string) {
+	var properties []IssueProperty
+	// 拿到了所有的需要迁移的——自定义的字段
+	//client.Table("dice_issue_property").Where("org_id = 1638 and scope_id = 1638 and property_issue_type = 'COMMON' and property_name not in('任务类型', '引入源')").
+	//	Find(&properties)
+	client.Debug().Table("dice_issue_property").Where("org_id = ?", orgID).Where("scope_id = ?", orgID).Where("property_issue_type = 'COMMON'").Where("property_name IN (?)", propertyNames).
+		Find(&properties)
+	// 集成测试
+	//'无效缺陷', '环境', '业务处室', '需要数据组'
+	//client.Table("dice_issue_property").Where("org_id = 2 and scope_id = 2 and property_issue_type = 'COMMON' and property_name in ('环境', '业务处室', '需要数据组') ").
+	//	Find(&properties)
+	for _, property := range properties {
+		var xuYao []IssueProperty
+		// 从第一个开始，拿到他能关联的三关系的数据,0-3个
+		//client.Table("dice_issue_property").Where("property_name = ?", property.PropertyName).
+		//	Where("scope_id = 1638 and org_id = 1638").Where("property_issue_type != 'COMMON' ").Find(&xuYao)
+		client.Table("dice_issue_property").Where("property_name = ?", property.PropertyName).
+			Where("org_id = ?", orgID).Where("scope_id = ?", orgID).Where("property_issue_type != 'COMMON' ").Find(&xuYao)
+		// 拿到关联的关系的 issue 的项目id
+		var projectIDs []IssuePropertyRelation
+		// map 记录common是否创建的有了，有了就不要创建了
+		existCommon := make(map[string]IssueProperty, 0)
+		for _, guanxi := range xuYao {
+			client.Table("dice_issue_property_relation").Where("property_id = ?", guanxi.ID).Group("project_id").Find(&projectIDs)
+			// 创建
+			// 要怎么实现，只创建一次 common？？？？？？？？？？？？用 map 去重
+			for _, commonAndThree := range projectIDs {
+				// 创建个 map 不要重复了
+				var result IssueProperty
+				abc := guanxi.PropertyName + ":" + "project" + ":" + strconv.FormatInt(commonAndThree.PropertyID, 10)
+				fmt.Println(abc)
+				if v, ok := existCommon[guanxi.PropertyName+":"+"project"+":"+strconv.FormatInt(commonAndThree.ProjectID, 10)]; ok {
```

Comment:
implement a func to make project property key

Reply:
okay, thank you for your suggestion. I tried a variable that might be easier